### ISSUE TITLE
Workaround for windows api/path when using neovim

### ISSUE THREAD

**Issue by shnaps** [Linus-Mussmaecher/rucola#13]
**Is your feature request related to a problem? Please describe.**
When using neovim on windows, there is an issue with the way neovim handles paths. Here is the issue from neovim repo https://github.com/neovim/neovim/issues/29495, it explains pretty much everything.

**Describe the solution you'd like**
When using windows machine, extend path and remove "\\\\?\\" when opening a note so it will provide fully qualified path.

Right now this issue added into a neovim backlog but that will probably take a long time to get fixed, so I though that adding a workaround  for this issue in rucola itself could be faster.

**Comment by Linus-Mussmaecher** [comment]
Hi,

thanks for the suggestion and sorry for the slow answer.
I have applied the requested fix by stripping this prefix from all paths - I don't think anything bad should happen on other OS (extensive testing on my Linux machine seems to confirm this).
The change has been pushed to the 'neovim' branch. Perhaps you (or some other Neovim + Windows user) could test it and confirm everything works now as expected so I can add this to the next release.

**Comment by shnaps** [comment]
Hi there!
That's all right, I wasn't waiting for an immediate fix or reply. I've checked your "neovim" branch. Still getting the same issue so I can only assume that neovim still tries to load the file the wrong way. Feel free to close the issue, we can only wait.
![image](https://github.com/user-attachments/assets/a916eaa2-56ba-4421-8fad-83e7df4bb026)
That's not a big deal, I was planning to walk away from neovim on windows for a couple of weeks now, managing all the dependencies is quite a hassle. Helix should work better in my case.

On a side note I have another question: have you considered adding Typst support as well? I find it a lot easier to write than *Tex stuff and it is not that verbose.

**Comment by shnaps** [comment]
Got some more testing 
![image](https://github.com/user-attachments/assets/114c3eb5-25cd-4f46-ad15-7888bad16359)
Is it expected to have a path like that?

**Comment by Linus-Mussmaecher** [comment]
Ah, I'm currently only replacing the `\\?\` when sending it to the external editing application (although apparently that doesn't work as your first testing shows).
I didn't think that the internal path might also contain it.

I have to admit, I'm not sure if it being there is a good or a bad thing - I suppose there must be a reason `.canonicalize()` adds it to the path, but I don't know enough about windows to know that reason. I think in general it turns off string parsing such as `.` or `..` paths, and extends their maximum length, but that doesn't seem to be necessary here.

**Comment by shnaps** [comment]
Ah, so that's expected behavior then? 
I don't get a bunch of things MC does either, haha. If you are curious, here is the link for their path stuff https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#dos-device-paths
I also moved to Helix and so far it works without any issues.

**Comment by Linus-Mussmaecher** [comment]
I suppose it is expected behavior in the sense that I'm simply calling Rust's [`.canonicalize()`](https://doc.rust-lang.org/beta/std/fs/fn.canonicalize.html) on the path, which explicitly mentions converting to 'extended length path syntax' on Windows.
Using these internally is rather advantageous, so I'm not sure if I want to move away from that.
I hoped simply removing the `\\?\` when outputting to external applications would help with this issue, but apparently it doesn't. Though admittedly this is difficult to test for me (I guess I could install Windows on a VM and do testing there).

Regarding winget, that's an interesting suggestion, but probably some work to understand and set up everything correctly. I'll look into it when I have the time.

**Comment by shnaps** [comment]
Fair enough, this is neovim issue in the end. It just doesn't support required API for win32 namespaces.

As for winget, it needs windows and that is it pretty much. I played with it for an hour, it was enough to create a simple package without creating pr for it.
